src/main/java/com/teamacronymcoders/epos/locks/keys/ItemLockKey.java

```diff
@@ -0,0 +1,54 @@
+package com.teamacronymcoders.epos.locks.keys;
+
+import com.teamacronymcoders.epos.api.locks.keys.ILockKey;
+import com.teamacronymcoders.epos.api.locks.keys.NBTLockKey;
+import java.util.Objects;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.CompoundNBT;
+
+public class ItemLockKey extends NBTLockKey {
+
+    @Nonnull
+    private Item item;
+
+    public ItemLockKey(@Nonnull Item item) {
+        this(item, null);
+    }
+
+    public ItemLockKey(@Nonnull Item item, @Nullable CompoundNBT nbt) {
+        super(nbt);
+        this.item = item;
+    }
+
+    @Nullable
+    public static ItemLockKey fromItemStack(@Nonnull ItemStack stack) {
+        return stack.isEmpty() ? null : new ItemLockKey(stack.getItem(), stack.getTag());
+        //TODO: Add case for block and for blockstate (or maybe just add a BlockStateLockKey)
```

Comment:
The idea here is for if we want to make it so that locks on items can be created from blocks, and I am open to either way.

- Way one lets you have different requirements for placing dirt and for breaking dirt, as the lock on dirt as a block won't resolve to an ItemLockKey as it is not an ItemStack/Item. Likewise, the BlockLockKey (which I will add before we merge this), won't be able to resolve from an ItemStack/Item as it is specific to blocks/items.
- Way two would make it so that you can lock using dirt as an item (which would also stop you from breaking it, basically how Reskillable has it). We then could also add a BlockLockKey that *adds* more requirements to the block form of dirt.

I think the first way makes more sense, *but* may be more annoying for pack devs as if they want the placing and breaking requirements to be the same they will have to declare the lock for both the item and the block.

Reply:
The issue is that Item -> Block aren't really great connections and a lot of Items that place blocks aren't always ItemBlock, that's why I believe they should be separate.